File: lab02/src/main.py
import time

def calculate(x):
    return x**2 - x**2 + x * 4 - x * 5 + x + x

def main():
    while True:
        try:
            n = int(input("Введите количество итераций (или букву для выхода): "))
        except ValueError:
            print("Введено не число. Завершение программы.")
            break

        if n <= 0:
            print("Введите положительное число!")
            continue

        start = time.perf_counter()

        result = 0
        for _ in range(n):
            result = calculate(2.0)

        end = time.perf_counter()
        duration_ms = (end - start) * 1000

        print(f"Выполнено итераций: {n}")
        print(f"Время выполнения: {duration_ms:.2f} мс")
        print("-" * 40)

if __name__ == "__main__":
    main()

File: lab02/src/main.cpp
#include <iostream>
#include <chrono>
#include <cmath>
#include <limits>

using namespace std;
using namespace chrono;

// Функция для вычисления выражения: x^2 - x^2 + x*4 - x*5 + x + x
double calculate(double x) {
    return pow(x, 2) - pow(x, 2) + x * 4 - x * 5 + x + x;
}

int main() {
    int n;

    while (true) {
        cout << "Введите количество итераций (или букву для выхода): ";

        // Проверка на ввод числа
        if (!(cin >> n)) {
            cout << "Введено не число. Завершение программы." << endl;
            break;
        }

        if (n <= 0) {
            cout << "Введите положительное число!" << endl;
            continue;
        }

        // Засекаем время СТАРТ
        auto start = high_resolution_clock::now();

        // Цикл на n итераций
        volatile double result = 0; // volatile чтобы компилятор не вырезал вычисления
        for (int i = 0; i < n; i++) {
            result = calculate(2.0); // x = 2 (можно подставить любое число)
        }

        // Засекаем время СТОП
        auto end = high_resolution_clock::now();
        auto duration = duration_cast<milliseconds>(end - start).count();

        // Вывод результата
        cout << "Выполнено итераций: " << n << endl;
        cout << "Время выполнения: " << duration << " мс" << endl;
        cout << "----------------------------------------" << endl;
    }

    return 0;
}
